scss/components/_labels.scss

```diff
@@ -43,3 +43,16 @@
   .label--yellow { background: $yellow; }
   .label--orange { background: $orange; }
   .label--red { background: $red; }
+
+  .label--outline {
+    background: none;
+    color: $gray;
+    border: 1px solid currentColor;
+
+    &.label--blue { color: $blue; }
+    &.label--green { color: $green; }
+    &.label--yellow { color: $yellow; }
```

Comment:
The outline label should be a new component, like `.outlineButton`. Avoid concatenation, especially with modifiers. 

But the main error here is:

```
.label--outline.label--yellow {
  background: $yellow;
  color: $yellow;
}
```

A fully yellow label.